### PR TITLE
MNTOR-1385 - request subscription scope, as some of our RPs are not yet allowed

### PR DESCRIPTION
**Pull request by rhelmer**
Request subscription scope when authenticating, as some of our RPs are not yet allowed.
This will fail early vs. just silently not working.

# References: 
Jira: MNTOR-1385

# How to test

Whether or not this works depends on the configuration of the environment on the FxA side, on investigating with SRE we found several client IDs registered for Fxa staging, most of which we are no longer using on the Monitor team.

SRE is adding the scope to our existing client IDs now, and we will work separately to deprecate any unused client IDs so there isn't confusing in the future. Additionally, SRE will enable this for production before we ship this feature (which is behind a flag).

# Checklist (Definition of Done)
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.

### CODE REVIEW

File: src/app/api/auth/[...nextauth]/route.ts [hunk 1]
```diff
@@ -59,7 +59,7 @@ export const authOptions: AuthOptions = {
       authorization: {
         url: AppConstants.OAUTH_AUTHORIZATION_URI,
         params: {
-          scope: "profile",
+          scope: "profile https://identity.mozilla.com/account/subscriptions",
```
Comment (pdehaan) [inline src/app/api/auth/[...nextauth]/route.ts:62]: Q: Should this be hardcoded URL, or should it be from an env var so we can toggle between stage/prod based on deployment?

Comment (rhelmer) [reply src/app/api/auth/[...nextauth]/route.ts:62]: It's actually the name of the scope (similar to `profile`) which is admittedly somewhat confusing :) I think it's being used as a URI not URL in this case.

Comment (rhelmer) [reply src/app/api/auth/[...nextauth]/route.ts:62]: https://mozilla.github.io/ecosystem-platform/api#tag/OAuth-Server-API-Overview/operation/postIntrospect for reference

**Comment by rhelmer** [comment]
BTW failing playwright test is expected as we're awaiting the `subscribers` scope to be allowed on the server-side:

image

**Review by Vinnl — APPROVED** [review]
I'm assuming the config will be added to FxA shortly? Otherwise it could be annoying that the Playwright tests will keep failing.

**Comment by rhelmer** [comment]
> I'm assuming the config will be added to FxA shortly? Otherwise it could be annoying that the Playwright tests will keep failing.

Yes, SRE is working on it now - I will hold off on landing this until that's in-place. Thanks!

**Comment by rhelmer** [comment]
SRE has enabled this for our stage client IDs, and the Playwright tests pass now (which makes sense because they use staging FxA) so it should presumably work from our dev/stage servers now too.

**Pull request merged**